### PR TITLE
Update Rust version

### PR DESCRIPTION
**Pull request by blaxill**
- Updates rust in Dockerfile
- updates packages as necessary
- fixes new clippy errors

Similar to #443 but slightly newer nightly (with full component set) and fixes to pass tests

edit: due to [this rustc ICE](https://github.com/rust-lang/rust/issues/64450) rustfmt example has to be disabled temporarily (`grep -v rustfmt` in run_examples) 

### Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

**Comment by blaxill** [comment]
Unfortunately this is running into this open rust error https://github.com/rust-lang/rust/issues/64450 

tl;dr: Updating our toolchain requires us to use an updated version of rustfmt (as its dependency rustc-ap-syntax_pos required an update to compile), which now imports rustc-ap-syntax_pos (and related crates) using an alias, which when targetting WebAssembly causes a rustc ICE.

File: scripts/run_examples [hunk 1]
```diff
@@ -11,7 +11,7 @@ source "$SCRIPTS_DIR/common"
 # Note that the chat example cannot be run because the client is interactive.
 # TODO(#462): Re-enable chat example, even if it may be just loading the server-side code and not
 # interact with it.
-readonly RUST_EXAMPLES="$(find examples -mindepth 2 -maxdepth 4 -type d -regex '.*/module.*/rust$' | cut -d'/' -f2 | uniq | grep -v chat)"
+readonly RUST_EXAMPLES="$(find examples -mindepth 2 -maxdepth 4 -type d -regex '.*/module.*/rust$' | cut -d'/' -f2 | uniq | grep -v chat | grep -v rustfmt)"
```
Comment (daviddrysdale) [inline scripts/run_examples]: Could we get a TODO(#NNN) to re-instate the rustfmt example?

Comment (blaxill) [reply scripts/run_examples]: Done

File: WORKSPACE [hunk 2]
```diff
@@ -255,7 +255,7 @@ rust_repository_set(
     name = "rust_linux_x86_64",
     exec_triple = "x86_64-unknown-linux-gnu",
     extra_target_triples = [],
-    iso_date = "2019-11-06",
+    iso_date = "2020-02-06",
```
Comment (tiziano88) [inline WORKSPACE]: Could you add a comment here and in the Dockerfile to make sure these values are kept in sync?

Comment (blaxill) [reply WORKSPACE]: Done

**Comment by tiziano88** [comment]
Note that the build time now has increased because the cached Docker image is no longer useful (see https://pantheon.corp.google.com/cloud-build/builds/34e7619c-fb81-4323-a0ab-38509de54151;step=1?project=oak-ci ).

@blaxill could you run `./scripts/docker_push` to make sure that the new version is uploaded?

**Comment by blaxill** [comment]
> @blaxill could you run `./scripts/docker_push` to make sure that the new version is uploaded?

Done, I think it should be updated now.
